app/src/main/cpp/skyline/common/android_settings.h

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include "settings.h"
+#include "jvm.h"
+
+namespace skyline {
+    /**
+     * @brief Handles settings on the android platform
+     * @note Lifetime of this class must not exceed the one of the JNIEnv contained inside ktSettings
+     */
+    class AndroidSettings final : public Settings {
+      private:
+        KtSettings ktSettings;
+
+      public:
+        /**
+         * @brief Will construct the underlying KtSettings object in-place
+         */
+        AndroidSettings(JNIEnv *env, jobject settingsInstance) : ktSettings(env, settingsInstance) {
+            Update();
+        }
+
+        /**
+         * @brief A constructor that takes ownership of the passed KtSettings object
```

Comment:
We know it's a ctor, you can drop that '@note Will take ownership of the passed KtSettings object'